components/inputs/input-textarea.js

```diff
@@ -0,0 +1,261 @@
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { formatNumber } from '@brightspace-ui/intl/lib/number.js';
+import { FormElementMixin } from '../form/form-element-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { inputLabelStyles } from './input-label-styles.js';
+import { inputStyles } from './input-styles.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+class InputTextArea extends FormElementMixin(SkeletonMixin(RtlMixin(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Indicates that the input value is invalid.
+			 */
+			ariaInvalid: { type: String, attribute: 'aria-invalid' },
+			/**
+			 * Additional information communicated in the aria-describedby on the input.
+			 */
+			description: { type: String, reflect: true },
+			/**
+			 * Disables the input.
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label for the input.
+			 */
+			label: { type: String },
+			/**
+			 * Hides the label visually (moves it to the input's "aria-label" attribute).
+			 */
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			/**
+			 * Maximum height of the input before scrolling.
+			 */
+			maxHeight: { type: String, attribute: 'max-height' },
+			/**
+			 * Imposes an upper character limit.
+			 */
+			maxlength: { type: Number },
+			/**
+			 * Minimum height of the input.
+			 */
+			minHeight: { type: String, attribute: 'min-height' },
+			/**
+			 * Imposes a lower character limit.
+			 */
+			minlength: { type: Number },
+			/**
+			 * Placeholder text.
+			 */
+			placeholder: { type: String },
+			/**
+			 * Indicates that a value is required.
+			 */
+			required: { type: Boolean, reflect: true },
+			/**
+			 * Value of the input.
+			 */
+			value: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ super.styles, inputStyles, inputLabelStyles, offscreenStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-input-label {
+				display: inline-block;
+				vertical-align: bottom;
+			}
+			:host .d2l-input-textarea-container {
+				max-width: 100%;
+				position: relative;
+			}
+			:host textarea {
+				height: 100%;
+				left: 0;
+				position: absolute;
+				resize: none;
+				top: 0;
+				z-index: 2;
```

Comment:
What does the default width end up being? Like does it fill any space that it's placed inside? I'm not sure what the correct thing would be here... but maybe something similar to the default `<textarea>` or maybe design has some thoughts. Same thoughts about the default height that I mentioned up above.

Reply:
The default width ends up being 100%, however the consumer can limit the width if they choose:
```html
<d2l-input-textarea style="width: 250px;">
```
However, it does not auto-grow horizontally.  So this doesn't quite work...
```html
<d2l-input-textarea style="display: inline-block;">
```
...because the mirror is a block and must match the `textarea` dimensions (it almost works).  We _might_ be able to enable horizontal auto-grow if that's required.

Reply:
I don't think horizontal autogrow would be useful. 100% width of its container seems correct, unless design thinks having a default max-width would make sense?

Reply:
I think defaulting to 100% width makes sense, that's the way it gets used most of the time these days.

Reply:
Awesome!  Thanks.  That makes the one issue in my other comment moot.  :smile: